napari/_qt/qt_viewer.py

```diff
@@ -108,9 +101,36 @@ def __init__(self, viewer):
             lambda event: self._update_palette(event.palette)
         )
         self.viewer.layers.events.reordered.connect(self._update_canvas)
+        self.viewer.dims.events.display.connect(self._update_camera)
 
         self.setAcceptDrops(True)
 
+    def _update_camera(self, event):
+        if sum(self.viewer.dims.display) == 3:
+            # Set a 3D camera
+            self.view.camera = TurntableCamera(name="TurntableCamera")
+            # Create an XYZaxis visual
+            self.axis = XYZAxis(parent=self.view)
+            self.axis.transform = STTransform(
+                translate=(50, 50), scale=(50, 50, 50, 1)
```

Comment:
where do these `(50, 50)` and `(50, 50, 50, 1)` numbers come from? do they relate the shape of the data, are they causing the zoom problems you mention?

Reply:
this just transforms the xyz  axis in red, green, blue color that is displayed on the right top
instead of showing at the origin i.e starting the axis from the top corner it translates it to 50, 50.
This is not causing any zoom problems. But this doesn't get deleted when the layer is deleted. This is a different kind of problem

Reply:
ah i get it, can discuss the value of that thing with @royerloic - i always find them slightly off putting - in the long run would be nice to have an option to turn them on / off but that can come later